ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -166,14 +185,24 @@ a database ( directory )
           }
           loadTaskTracker.update(functionsTracker);
           functionsTracker.debugLog("functions");
+
+          // Add ReplStateLogTask as function load tasks are always fully added in current cycle
+          loadTaskTracker.updateTaskCount(createFunctionReplLogTask(functionsTracker.tasks(),
+                  iterator.replLogger(), loadFunction.functionName()));
```

Comment:
the repl log tasks are not going to go inside the trackers, they will go in the respective classes responsible for loading tables/functions/partitions/databases etc. the correct dependency of these tasks will also be added from within the task generators for objects. 

adding this code in replloadtask is making it more unreadable and complicated rather than having them within the classes that produce respective tasks.